custom_components/google_home/api.py

```diff
@@ -113,51 +113,63 @@ async def get_alarms_and_timers(self, device):
-                self.google_devices = []
-            elif response.status == HTTP_NOT_FOUND:
-                device.available = False
-                _LOGGER.debug(
-                    (
-                        "Failed to fetch data from %s, API returned %d. "
-                        "The device(hardware='%s') is not Google Home "
-                        "compatable and has no alarms/timers."
-                    ),
-                    device.name,
-                    response.status,
-                    device.hardware,
-                )
-            else:
-                _LOGGER.error(
-                    "Failed to fetch %s data, API returned %d: %s",
-                    device.name,
-                    response.status,
-                    response,
-                )
 
-        if resp:
-            if JSON_TIMER in resp or JSON_ALARM in resp:
-                device.set_timers(resp.get(JSON_TIMER))
-                device.set_alarms(resp.get(JSON_ALARM))
-            else:
-                _LOGGER.error(
-                    "For device %s - %s",
-                    device.name,
-                    API_RETURNED_UNKNOWN,
-                )
+        try:
+            async with self._session.get(
+                url, headers=HEADERS, timeout=TIMEOUT
+            ) as response:
+                if response.status == HTTP_OK:
+                    resp = await response.json()
+                elif response.status == HTTP_UNAUTHORIZED:
+                    # If token is invalid - force reload homegraph providing new token
+                    # and rerun the task.
+                    _LOGGER.debug(
+                        (
+                            "Failed to fetch data from %s due to invalid token. "
+                            "Will refresh the token and try again."
+                        ),
+                        device.name,
+                    )
+                    # We need to retry the update task instead of just cleaning the list
+                    self.google_devices = []
+                elif response.status == HTTP_NOT_FOUND:
+                    device.available = False
+                    _LOGGER.debug(
+                        (
+                            "Failed to fetch data from %s, API returned %d. "
+                            "The device(hardware='%s') is not Google Home "
+                            "compatable and has no alarms/timers."
+                        ),
+                        device.name,
+                        response.status,
+                        device.hardware,
+                    )
+                else:
+                    _LOGGER.error(
+                        "Failed to fetch %s data, API returned %d: %s",
+                        device.name,
+                        response.status,
+                        response,
+                    )
+        except Exception as exc:  # pylint: disable=broad-except
```

Comment:
`ex` is more common here. If pylint complains about it, in `pyproject.toml` add:
```
[tool.pylint.basic]
good-names = ["ex"]
```

Reply:
Also catching all possible exceptions in this large block of code is really too broad. I think you only need `aiohttp.ClientError` here.

Reply:
I know it is bad to catch broad, but i specifically thought that in this case it will be pretty useful as a debug tool for us.

But it is possible to use another client library then `aiohttp`?

Reply:
No you are right, it is not compatible, I forgot that `aiohttp` is a little bit different then others in its api.